### PR TITLE
Add input to filter by program title

### PR DESCRIPTION
**Pull request by bseverino**
### [MST-460](https://openedx.atlassian.net/browse/MST-460)

Adds the ability to filter programs by title, and displays an error when the title doesn't exist (instead of updating program data with an empty array).

---
Screen Shot 2020-10-19 at 1 16 33 PM
---
Screen Shot 2020-10-19 at 1 16 22 PM
---
Screen Shot 2020-10-19 at 1 16 50 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "@edx/frontend-component-footer": "^10.0.11",
     "@edx/frontend-component-header": "^2.0.5",
     "@edx/frontend-platform": "^1.5.3",
-    "@edx/paragon": "^7.2.0",
+    "@edx/paragon": "^12.0.4",
```
Comment (schenedx) [inline package.json:32]: I am a bit worried about this version jump. It this upgrade introduce any non-related UI changes on the program-console?

File: src/console/ConsolePage.test.jsx [hunk 2]
```diff
@@ -254,15 +261,125 @@ describe('ConsolePage', () => {
...
 
     expect(mock).toHaveBeenCalled();
   });
 
+  it('calls the filterPrograms function on form submit', () => {
+    const mock = jest.fn();
+
+    expect(mock).not.toHaveBeenCalled();
+
+    const apiData = [{
```
Comment (schenedx) [inline src/console/ConsolePage.test.jsx:279]: Seems like this `const` is defined to be the same for all your tests, can we refactor?

**Review by schenedx — APPROVED** [review]
My comments are minor. Changes looks nice. Please consider demo at the next earliest opportunity

File: package.json (hunk 1 above)
Comment (bseverino) [reply package.json:32]: I will double check.

Comment (bseverino) [reply package.json:32]: As far as I can tell, the components we're using (which were just Collapsible and StatusAlert before) aren't affected by any breaking changes.

**Pull request merged**